File: src/com/colacelli/irclib/connection/listeners/OnPrivateMessageListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.connection.Connection;
import com.colacelli.irclib.messages.PrivateMessage;

public interface OnPrivateMessageListener extends Listener {
    Type TYPE = Type.PRIVATE_MESSAGE;

    void onPrivateMessage(Connection connection, PrivateMessage message);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/main/kotlin/com/colacelli/irclib/connection/listeners/OnPrivateMessageListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.connection.Connection
import com.colacelli.irclib.messages.PrivateMessage

interface OnPrivateMessageListener : Listener {
    fun onPrivateMessage(connection: Connection, message: PrivateMessage)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.PRIVATE_MESSAGE
    }
}
